Add pull request guidelines to CONTRIBUTING.md (#17660)
rstudio/rstudio: CONTRIBUTING.md

3. Submit a [pull request](https://help.github.com/articles/using-pull-requests).

We'll try to be as responsive as possible in reviewing and accepting pull requests. We appreciate your contributions very much!

## Pull Request Guidelines

Reviewing pull requests (PRs) takes real time from a small team, so we ask that every contribution meets the bar described below. PRs that don't will be closed.

### Build and Test Locally

Every pull request is expected to have been built and tested locally before it is opened. In the PR description, please include:

- **What you tested**: the scenarios, workflows, or behaviors you verified, including any edge cases you considered
- **Which platforms you tested on**: macOS, Windows, or Linux

If you were unable to test on a particular platform, say so explicitly. We'd much rather know up front than discover it during review or after merge.

### Limit on Open Pull Requests

> External contributors may have at most two open pull requests at a time. Additional PRs opened while two are already open will be closed without review.

Reviewer: ```suggestion
We ask that external contributors keep at most two open pull requests at a time. Additional PRs opened while two are already open will be closed without review, but may be re-opened at later time.
```

> Once a contributor has established a track record of thoughtful, high-quality contributions, we're happy to lift this limit on a case-by-case basis. Just ask.

Reviewer: ```suggestion
Once a contributor has established a track record of thoughtful, high-quality contributions, we're happy to lift this limit on a case-by-case basis.
```

### AI-Assisted Contributions

Using AI tools to help write code or PR descriptions is not prohibited, but the responsibility for the contribution is yours. Before submitting, please:

- Read every line of the change and make sure you understand it
- Confirm the change actually addresses a real problem, not a hypothetical one inferred from the code
- Verify the change fits the surrounding code style and conventions

> - Make sure the PR description matches what the diff actually does

Reviewer: I'm not 100% set on this one, what do you think?
```suggestion
- Make sure the PR description matches what the diff actually does
- Include an acknowledgement that AI was used to help write this code 
```

Author: I think it's a given that AI is being used on most if not all PRs. I'd almost be more worried about a change made without AI at this point.
